lib/saml.js

```diff
@@ -180,33 +180,58 @@ SAML.prototype.certToPEM = function (cert) {
 };
 
 SAML.prototype.checkSAMLStatus = function (xmlDomDoc) {
-	var status = {StatusCodeValue:null, StatusMessage:null, StatusDetail:null}
+	var status = {StatusCodeValue:null, StatusMessage:null, StatusDetail:null}, statusCodeValueNode = null, statusMessageNode = null, statusDetailNode = null;
 
-	var statusCodeValueNode = xmlCrypto.xpath(xmlDomDoc, "//*[local-name(.)='StatusCode']")[0];
-	if(statusCodeValueNode)
-	{
+	try{
+    statusCodeValueNode = xmlCrypto.xpath(xmlDomDoc, "//*[local-name(.)='StatusCode']")[0];
+  }catch(err){
+    // Error Handling
+  }
+
+	if(statusCodeValueNode = true || typeof statusCodeValueNode != 'undefined'){
```

Comment:
Sorry, missed this on the first review pass.
This `if(typeof statusCodeValueNode != 'undefined')` is sufficient.

If it's not undefined,  we set the value.

(Also, minor bug, you used a single `=` when comparing to true. That is assignment, not comparison), but it doesn't matter since it ill be removed.

I can merge after this